libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewInstanceState.java

```diff
@@ -0,0 +1,76 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.Nullable;
+import android.support.design.widget.BottomSheetBehavior;
+
+class NavigationViewInstanceState implements Parcelable {
+  private Integer bottomSheetBehaviorState;
+  private int recenterButtonVisibility;
+  private boolean instructionViewVisible;
+
+  NavigationViewInstanceState(@Nullable BottomSheetBehavior bottomSheetBehavior, int recenterButtonVisibility,
```

Comment:
Have we considered passing the state right away? Instead of passing the whole object and obtaining the state within the constructor.

Reply:
@Guardiola31337 I just thought it was nice to get the null check out of the `NavigationView`

Reply:
@devotaaabel 
Oh yeah, I meant instead of passing in `BottomSheetBehavior` use an `Integer` so we don't need to call `bottomSheetBehavior.getState()`. Does that make sense?
Also, if not initialized `bottomSheetBehaviorState` should be `null` so no need to set it to `null` in the `else` branch, right?